src/components/productList/ProductList.component.tsx

```diff
@@ -14,79 +14,72 @@ import styles from './ProductList.module.css';
 
 const ProductsList: React.FC = () => {
     const [currentPage, setCurrentPage] = useState(1);
-    const [filteredProducts, setFilteredProducts] = useState<Product[]>([]);
+    const [products, setProducts] = useState<Product[]>([]);
+    const [categories, setCategories] = useState<Category[]>([]);
     const [selectedCategories, setSelectedCategories] = useState<Category[]>([]);
     const [sortOption, setSortOption] = useState<SortOption>(SortOption.PRICE_HIGH_TO_LOW);
     const [searchQuery, setSearchQuery] = useState<string>('');
     const [searchInput, setSearchInput] = useState<string>('');
+    const [totalPages, setTotalPages] = useState(0);
```

Comment:
Now since the pagination, sorting and filtering are done on the backend, you should use query params for handling these changes. The idea is simple: User should be able to share/access the page and see the filtered list of products.

Learn more: https://reactrouter.com/en/main/hooks/use-search-params